lib/puppet/functions/reverse_each.rb

```diff
@@ -0,0 +1,88 @@
+# Reverses the order of the elements of something that is iterable and optionally runs a
```

Comment:
We must check if it is ok to just define the 4.x function, or if we have to do a 3.x version only to get the documentation generated - ping @nfagerlund